### PR TITLE
Make separate browser bundles with the Emscripten browser-only build

### PR DESCRIPTION
**Pull request by nrabinowitz**
Refactors the build scripts to create separate browser bundles, using an Emscripten browser environment build to cut out the Node-specific code.

The `yarn dist` command now makes the following bundles:

- `dist/h3-js.js` - CommonJS multi-platform bundle
- `dist/h3-js.es.js` - ES6 multi-platform bundle
- `dist/h3-js.umd.js` - UMD multi-platform bundle
- `dist/browser/h3-js.js` - CommonJS browser-only bundle
- `dist/browser/h3-js.es.js` - ES6 browser-only bundle

The `browser` field remaps the multi-platform bundles to the browser bundles for tools like Webpack and Rollup.

I still need to test that this all works as expected; I'm hoping that @Pessimistress and @macrigiuseppe might be able to confirm that these bundles work in the Kepler and Deck.gl builds.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,8 +31,8 @@
     "bundle-umd": "microbundle --name h3 --format=umd",
     "bundle-cjs": "microbundle --format=cjs --no-compress",
     "bundle-es": "microbundle --format=es --no-compress",
-    "bundle-cjs-browser": "microbundle -o dist/browser --format=cjs --no-compress --alias ./out/libh3=./dist/libh3-browser",
-    "bundle-es-browser": "microbundle -o dist/browser --format=es --no-compress --alias ./out/libh3=./dist/libh3-browser",
+    "bundle-cjs-browser": "microbundle -o dist/browser --format=cjs --no-compress --alias ../out/libh3=$PWD/dist/libh3-browser",
+    "bundle-es-browser": "microbundle -o dist/browser --format=es --no-compress --alias ../out/libh3=$PWD/dist/libh3-browser",
```
Comment (isaacbrodsky) [inline package.json]: Will `--alias ../out/libh3=$PWD/dist/libh3-browser` behave unexpectedly when `$PWD` contains spaces?

Comment (nrabinowitz) [reply package.json]: ...Good question. Not sure what the best practice is here - though note that this only needs to run on developer machines and Travis, so mistakes here won't be an issue for end users.

Comment (nrabinowitz) [reply package.json]: Updated to `$(printf '%q' $PWD)` to escape.

Comment (isaacbrodsky) [reply package.json]: Is that different from `\"../out/libh3=$PWD/dist/libh3-browser\"`?

**Pull request merged**